app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx

```diff
@@ -0,0 +1,99 @@
+/** @file Provides an {@link AuthService} which consists of an underyling {@link Cognito} API
+ * wrapper, along with some convenience callbacks to make URL redirects for the authentication flows
+ * work with Electron. */
+import * as cognito from "./cognito";
+import * as authConfig from "./config";
+import * as config from "../config";
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const BASE_AMPLIFY_CONFIG: Partial<authConfig.AmplifyConfig> = {
+    region: authConfig.AWS_REGION,
+    scope: authConfig.OAUTH_SCOPES,
+    responseType: authConfig.OAUTH_RESPONSE_TYPE,
+};
+
+/** Collection of configuration details for Amplify user pools, sorted by deployment environment. */
+const AMPLIFY_CONFIGS = {
+    /** Configuration for @pbuchu's Cognito user pool. */
+    pbuchu: {
+        userPoolId: "eu-west-1_jSF1RbgPK",
+        userPoolWebClientId: "1bnib0jfon3aqc5g3lkia2infr",
+        domain: "pb-enso-domain.auth.eu-west-1.amazoncognito.com",
+        ...BASE_AMPLIFY_CONFIG,
+    },
+    /** Configuration for the production Cognito user pool. */
+    production: {
+        userPoolId: "eu-west-1_9Kycu2SbD",
+        userPoolWebClientId: "4j9bfs8e7415erf82l129v0qhe",
+        domain: "production-enso-domain.auth.eu-west-1.amazoncognito.com",
+        ...BASE_AMPLIFY_CONFIG,
+    },
+};
+
+
+
+// ==================
+// === AuthConfig ===
+// ==================
+
+/** Configuration for the authentication service. */
+export interface AuthConfig {
+    /** Whether the application is running on a desktop (i.e., versus in the Cloud). */
+    runningOnDesktop: boolean;
+}
+
+
+
+// ===================
+// === AuthService ===
+// ===================
+
+/** API for the authentication service. */
+export interface AuthService {
+    /** @see {@link Cognito} */
+    cognito: cognito.Cognito;
+}
+
+/** Creates an instance of the authentication service.
+ *
+ * # Warning
+ *
+ * This function should only be called once, and the returned service should be used throughout the
+ * application. This is because it performs global configuration of the Amplify library.
+ *
+ * @param authConfig - Configuration for the authentication service.
+ * @returns An instance of the authentication service. */
+export const initAuthService = (authConfig: AuthConfig): AuthService => {
+    const { runningOnDesktop } = authConfig;
+
+    const amplifyConfig = loadAmplifyConfig(runningOnDesktop);
+    const cognitoClient = new cognito.CognitoImpl(
+        amplifyConfig
+    );
+
```

Comment:
1. why these newlines? :( 
2. There is something wrong with prettier configuration. Both line 75 and 79 should be one liners. Now they spread over several lines.

Reply:
Personally I prefer separating:
1. argument destructuring
2. logic
3. returns
but I'll stop doing that.

As for the prettier, yes I made some changes and forgot to re-run. Will fix.

Reply:
@wdanilo has said elsewhere that it's against our style guide to destructure in the parameter list